recovery/main.cpp

```diff
@@ -81,9 +81,8 @@ QString findRecoveryDrive()
 
     foreach (QString devname, list)
     {
-        /* Only search first partition and partitionless devices. Skip virtual devices (such as ramdisk) */
-        if ((devname.right(1).at(0).isDigit() && !devname.endsWith("1"))
```

Comment:
Ahhhh, now I finally see why the existing "partitionless devices" check was failing! :laughing: 
It was checking that the devname either doesn't end with a digit (e.g. `sda` for a partitionless USB drive), or it ends with 1 (e.g. `sda1` for a partitioned USB drive). This ensures that it doesn't erroneously "find" `recovery.rfs` e.g. on `/dev/sda3`.

But when we're booting from an SD-card, both  the partitionless device (i.e. `mmcblk0`) **and** the partitioned device (i.e. `mmcblk0p1`) both end in a digit!

So @procount I wonder if a better fix would be to change it to
```
if ((devname.right(1).at(0).isDigit() && !(devname.endsWith("1") || devname.endsWith("0")))
    || QFile::symLinkTarget("/sys/class/block/"+devname).contains("/devices/virtual/"))
```
:question: 

(I'm afraid I don't have tine to test this myself)